modules/aws/rds.go

```diff
@@ -86,15 +86,91 @@ func GetParameterValueForParameterOfRdsInstance(t testing.TestingT, parameterNam
 	return parameterValue
 }
 
+//------------------------------------------------------------------------------------------------------------------------------------------------
 // GetParameterValueForParameterOfRdsInstanceE gets the value of the parameter name specified for the RDS instance in the given region.
+// modified to use previos token and page through all parameters
 func GetParameterValueForParameterOfRdsInstanceE(t testing.TestingT, parameterName string, dbInstanceID string, awsRegion string) (string, error) {
-	output := GetAllParametersOfRdsInstance(t, dbInstanceID, awsRegion)
-	for _, parameter := range output {
-		if aws.StringValue(parameter.ParameterName) == parameterName {
-			return aws.StringValue(parameter.ParameterValue), nil
+	lastmarker := ""			// var to store the returned Marker value
+	morepages := true 			// initialize morepages to indicate more pages need to be pulled
+	pagecnt := 1				// initialize pagecnt to be used to limit the number if loops in the event the parameter cannot be found
+	for morepages == true {
+		//  *** on the first pass, the lastmarker is nil, if additional records exist, the api call will return a marker. cnt holds the number of items in output
+		output, marker, cnt := GetParametersOfRdsInstance(t, dbInstanceID, awsRegion, lastmarker)
+		// store the new marker to a local var
+		lastmarker = marker
+		//  check if the number of records returned are less than 100 or if a marker was not returned to have function stop on this current page
+		if (cnt < 100) || (marker == "") {
+			morepages = false
 		}
```

Comment:
When `cnt` is less than 100, is `marker` set? If not, you can probably simplify this to just `if marker == ""`.